File: object_initializer.cpp
// includes
#include "object_initializer.h"

namespace generic_protocol
{

// objects

void initialize( SessionInfo * res
    , uint32_t user_id
    , uint32_t start_time
    , uint32_t expiration_time
 )
{
    res->user_id = user_id;
    res->start_time = start_time;
    res->expiration_time = expiration_time;
}

// base messages

void initialize( ForwardMessage * res
 )
{
    // no base class
}

void initialize( BackwardMessage * res
 )
{
    // no base class
}

void initialize( Request * res
    , const std::string &  session_id
 )
{
    // base class
    initialize( static_cast<ForwardMessage*>( res ) );

    res->session_id = session_id;
}

// messages

void initialize( ErrorResponse * res
    , ErrorResponse_type_e type
    , const std::string &  descr
 )
{
    // base class
    initialize( static_cast<BackwardMessage*>( res ) );

    res->type = type;
    res->descr = descr;
}

void initialize( AuthenticateRequest * res
    , const std::string &  user_login
    , const std::string &  password
 )
{
    // base class
    initialize( static_cast<ForwardMessage*>( res ) );

    res->user_login = user_login;
    res->password = password;
}

void initialize( AuthenticateAltRequest * res
    , uint32_t user_id
    , const std::string &  password
 )
{
    // base class
    initialize( static_cast<ForwardMessage*>( res ) );

    res->user_id = user_id;
    res->password = password;
}

void initialize( AuthenticateResponse * res
    , const std::string &  session_id
 )
{
    // base class
    initialize( static_cast<BackwardMessage*>( res ) );

    res->session_id = session_id;
}

void initialize( CloseSessionRequest * res
    , const std::string &  session_id
 )
{
    // base class
    initialize( static_cast<ForwardMessage*>( res ) );

    res->session_id = session_id;
}

void initialize( CloseSessionResponse * res
 )
{
    // base class
    initialize( static_cast<BackwardMessage*>( res ) );

}

void initialize( GetUserIdRequest * res
    , const std::string &  base_class_param_1
    , const std::string &  user_login
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->user_login = user_login;
}

void initialize( GetUserIdResponse * res
    , uint32_t user_id
 )
{
    // base class
    initialize( static_cast<BackwardMessage*>( res ) );

    res->user_id = user_id;
}

void initialize( GetSessionInfoRequest * res
    , const std::string &  base_class_param_1
    , const std::string &  id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->id = id;
}

void initialize( GetSessionInfoResponse * res
    , const SessionInfo &  session_info
 )
{
    // base class
    initialize( static_cast<BackwardMessage*>( res ) );

    res->session_info = session_info;
}

// messages (constructors)

ErrorResponse * create_ErrorResponse(
    ErrorResponse_type_e type
    , const std::string &  descr
 )
{
    auto * res = new ErrorResponse;

    initialize( res, type, descr );

    return res;
}

AuthenticateRequest * create_AuthenticateRequest(
    const std::string &  user_login
    , const std::string &  password
 )
{
    auto * res = new AuthenticateRequest;

    initialize( res, user_login, password );

    return res;
}

AuthenticateAltRequest * create_AuthenticateAltRequest(
    uint32_t user_id
    , const std::string &  password
 )
{
    auto * res = new AuthenticateAltRequest;

    initialize( res, user_id, password );

    return res;
}

AuthenticateResponse * create_AuthenticateResponse(
    const std::string &  session_id
 )
{
    auto * res = new AuthenticateResponse;

    initialize( res, session_id );

    return res;
}

CloseSessionRequest * create_CloseSessionRequest(
    const std::string &  session_id
 )
{
    auto * res = new CloseSessionRequest;

    initialize( res, session_id );

    return res;
}

CloseSessionResponse * create_CloseSessionResponse(
 )
{
    auto * res = new CloseSessionResponse;

    initialize( res );

    return res;
}

GetUserIdRequest * create_GetUserIdRequest(
    const std::string &  base_class_param_1
    , const std::string &  user_login
 )
{
    auto * res = new GetUserIdRequest;

    initialize( res, base_class_param_1, user_login );

    return res;
}

GetUserIdResponse * create_GetUserIdResponse(
    uint32_t user_id
 )
{
    auto * res = new GetUserIdResponse;

    initialize( res, user_id );

    return res;
}

GetSessionInfoRequest * create_GetSessionInfoRequest(
    const std::string &  base_class_param_1
    , const std::string &  id
 )
{
    auto * res = new GetSessionInfoRequest;

    initialize( res, base_class_param_1, id );

    return res;
}

GetSessionInfoResponse * create_GetSessionInfoResponse(
    const SessionInfo &  session_info
 )
{
    auto * res = new GetSessionInfoResponse;

    initialize( res, session_info );

    return res;
}

} // namespace generic_protocol

File: object_initializer.php
<?php

namespace generic_protocol;


// includes
require_once 'protocol.php';

// objects

function initialize__SessionInfo( & $res
    , $user_id // uint32_t
    , $start_time // uint32_t
    , $expiration_time // uint32_t
 )
{
    $res->user_id = $user_id;
    $res->start_time = $start_time;
    $res->expiration_time = $expiration_time;
}

// base messages

function initialize__ForwardMessage( & $res
 )
{
    // no base class
}

function initialize__BackwardMessage( & $res
 )
{
    // no base class
}

function initialize__Request( & $res
    , $session_id // string
 )
{
    // base class
    initialize__ForwardMessage( $res );

    $res->session_id = $session_id;
}

// messages

function initialize__ErrorResponse( & $res
    , $type // ErrorResponse_type_e
    , $descr // string
 )
{
    // base class
    initialize__BackwardMessage( $res );

    $res->type = $type;
    $res->descr = $descr;
}

function initialize__AuthenticateRequest( & $res
    , $user_login // string
    , $password // string
 )
{
    // base class
    initialize__ForwardMessage( $res );

    $res->user_login = $user_login;
    $res->password = $password;
}

function initialize__AuthenticateAltRequest( & $res
    , $user_id // uint32_t
    , $password // string
 )
{
    // base class
    initialize__ForwardMessage( $res );

    $res->user_id = $user_id;
    $res->password = $password;
}

function initialize__AuthenticateResponse( & $res
    , $session_id // string
 )
{
    // base class
    initialize__BackwardMessage( $res );

    $res->session_id = $session_id;
}

function initialize__CloseSessionRequest( & $res
    , $session_id // string
 )
{
    // base class
    initialize__ForwardMessage( $res );

    $res->session_id = $session_id;
}

function initialize__CloseSessionResponse( & $res
 )
{
    // base class
    initialize__BackwardMessage( $res );

}

function initialize__GetUserIdRequest( & $res
    , $base_class_param_1 // string
    , $user_login // string
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->user_login = $user_login;
}

function initialize__GetUserIdResponse( & $res
    , $user_id // uint32_t
 )
{
    // base class
    initialize__BackwardMessage( $res );

    $res->user_id = $user_id;
}

function initialize__GetSessionInfoRequest( & $res
    , $base_class_param_1 // string
    , $id // string
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->id = $id;
}

function initialize__GetSessionInfoResponse( & $res
    , $session_info // SessionInfo
 )
{
    // base class
    initialize__BackwardMessage( $res );

    $res->session_info = $session_info;
}

// objects (constructors)

function create__SessionInfo(
    $user_id // uint32_t
    , $start_time // uint32_t
    , $expiration_time // uint32_t
 )
{
    $res = new SessionInfo;

    initialize__SessionInfo( $res, $user_id, $start_time, $expiration_time );

    return $res;
}

// messages (constructors)

function create__ErrorResponse(
    $type // ErrorResponse_type_e
    , $descr // string
 )
{
    $res = new ErrorResponse;

    initialize__ErrorResponse( $res, $type, $descr );

    return $res;
}

function create__AuthenticateRequest(
    $user_login // string
    , $password // string
 )
{
    $res = new AuthenticateRequest;

    initialize__AuthenticateRequest( $res, $user_login, $password );

    return $res;
}

function create__AuthenticateAltRequest(
    $user_id // uint32_t
    , $password // string
 )
{
    $res = new AuthenticateAltRequest;

    initialize__AuthenticateAltRequest( $res, $user_id, $password );

    return $res;
}

function create__AuthenticateResponse(
    $session_id // string
 )
{
    $res = new AuthenticateResponse;

    initialize__AuthenticateResponse( $res, $session_id );

    return $res;
}

function create__CloseSessionRequest(
    $session_id // string
 )
{
    $res = new CloseSessionRequest;

    initialize__CloseSessionRequest( $res, $session_id );

    return $res;
}

function create__CloseSessionResponse(
 )
{
    $res = new CloseSessionResponse;

    initialize__CloseSessionResponse( $res );

    return $res;
}

function create__GetUserIdRequest(
    $base_class_param_1 // string
    , $user_login // string
 )
{
    $res = new GetUserIdRequest;

    initialize__GetUserIdRequest( $res, $base_class_param_1, $user_login );

    return $res;
}

function create__GetUserIdResponse(
    $user_id // uint32_t
 )
{
    $res = new GetUserIdResponse;

    initialize__GetUserIdResponse( $res, $user_id );

    return $res;
}

function create__GetSessionInfoRequest(
    $base_class_param_1 // string
    , $id // string
 )
{
    $res = new GetSessionInfoRequest;

    initialize__GetSessionInfoRequest( $res, $base_class_param_1, $id );

    return $res;
}

function create__GetSessionInfoResponse(
    $session_info // SessionInfo
 )
{
    $res = new GetSessionInfoResponse;

    initialize__GetSessionInfoResponse( $res, $session_info );

    return $res;
}

# namespace_end generic_protocol


?>
